### DISCUSSION TITLE
How can I improve the quality by input part of the script?

### DISCUSSION THREAD

**Discussion by iTula** [Purfview/whisper-standalone-win#259] [Q&A]
We are using this app to transcribe a meeting. Some or part of the speeches in meeting are based on a script. How can I improve the quality of transcription using the script?

**Comment by Tamnac** [comment]
pass it in as the `--initial_prompt` (won't work with distil models afaik)

**Comment by iTula** [comment]
How can I do that? Can you give me an example? Thank you a lot.
发自我的 iPhone

在 2024年8月4日，03:49，Tamnac ***@***.***> 写道：

﻿

pass it in as the --initial_prompt (won't work with distil models afaik)

**Reply by Tamnac** [reply]
I assume you know how to use the cli, so you just do something like `.\faster-whisper-xxl.exe <other options> --initial_prompt "<script>".

That said, the prompt is more for like context/topic, if the script is included in the speeches word for word, you might be better of passing in the transcript + script into something like chatgpt and asking it to combine them.

**Reply by ClaireCJS** [reply]
> I assume you know how to use the cli, so you just do something like `.\faster-whisper-xxl.exe --initial_prompt "<script>".
> 
> That said, the prompt is more for like context/topic, if the script is included in the speeches word for word, you might be better of passing in the transcript + script into something like chatgpt and asking it to combine them.


I'm in a slightly different  situation and wondering if you could offer your advice.

A lot of music has a TXT download of lyrics, but no SRT/LRC.

The AI often guesses the wrong word when making SRC/LRC from scratch.

Is there any way to not prompt it, but prime it with the TXT file, so that it doesn't make incorrect guesses?  Basically, if I have the lyrics already, I just want it to add the timing.

**Reply by Tamnac** [reply]
I think you'll have to do the second suggestion. Make a transcript using this, then give it with timestamps  + your txt file to an llm like chatgpt  and ask it to combine the two using the txt as the content and the transcript for timestamps

**Reply by ClaireCJS** [reply]
> I think you'll have to do the second suggestion. Make a transcript using this, then give it with timestamps + your txt file to an llm like chatgpt and ask it to combine the two using the txt as the content and the transcript for timestamps


That is a shame. Could there be a way to add a TXT to it in advance, and perhaps it could then be turned into a prompt along the lines of "using these lyrics, do <whatever it is the current transcription prompt is>"?

**Comment by iTula** [comment]
Thanks, I will try that out.
________________________________
发件人: Tamnac ***@***.***>
发送时间: 2024年8月6日 19:40
收件人: Purfview/whisper-standalone-win ***@***.***>
抄送: iTula ***@***.***>; Author ***@***.***>
主题: Re: [Purfview/whisper-standalone-win] How can I improve the quality by input part of the script? (Discussion #259)


I assume you know how to use the cli, so you just do something like `.\faster-whisper-xxl.exe --initial_prompt "<script>".

That said, the prompt is more for like context/topic, if the script is included in the speeches word for word, you might be better of passing in the transcript + script into something like chatgpt and asking it to combine them.

―
